drivers/watchdog/iwdg_stm32.c

```diff
@@ -61,29 +62,35 @@ static void iwdg_stm32_convert_timeout(u32_t timeout,
 	*reload = (u32_t)(m_timeout / divider) - 1;
 }
 
-static void iwdg_stm32_enable(struct device *dev)
+static int iwdg_stm32_setup(struct device *dev, u8_t options)
 {
 	IWDG_TypeDef *iwdg = IWDG_STM32_STRUCT(dev);
 
+	/* Deactivate running when debugger is attached. */
+	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
+		LL_APB1_GRP2_EnableClock(LL_APB1_GRP2_PERIPH_DBGMCU);
+		LL_DBGMCU_APB1_GRP1_FreezePeriph(LL_DBGMCU_APB1_GRP1_IWDG_STOP);
+	}
+
```

Comment:
Add check for WDT_OPT_PAUSE_IN_SLEEP and return ENOTSUP in that case

Reply:
Done